### PR TITLE
MIME4J-289 Upgrade libs

### PR DESCRIPTION
**Pull request by trantienduchn**
```
      1 [INFO]   com.google.guava:guava .............................. 18.0 -> 28.1-jre
      9 [INFO]   commons-io:commons-io ..................................... 2.4 -> 2.6
      9 [INFO]   commons-logging:commons-logging ......................... 1.1.1 -> 1.2
      9 [INFO]   junit:junit ...................................... 4.10 -> 4.12
      9 [INFO]   log4j:log4j ......................................... 1.2.14 -> 1.2.17
     18 [INFO]   org.apache.maven.doxia:doxia-core ......................... 1.2 -> 1.9
      9 [INFO]   org.assertj:assertj-core ............................. 1.7.1 -> 3.13.2
      9 [INFO]   org.mockito:mockito-core .............................. 1.9.5 -> 3.0.0
      1 [INFO]   org.slf4j:slf4j-api ............................ 1.7.7 -> 1.7.18
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -72,7 +72,7 @@
         <commons-logging.version>1.2</commons-logging.version>
         <log4j.version>1.2.17</log4j.version>
         <junit.version>4.12</junit.version>
-        <mockito.version>1.9.5</mockito.version>
+        <mockito.version>3.0.0</mockito.version>
```
Comment (Arsnael) [inline pom.xml]: The versions described in the commit and the ones changed are different

**Comment by chibenwa** [comment]
```
[ERROR] Failures: 
[ERROR]   MessageCharsetLenientTest.testLenientCharsetHandling:117 Charset:iso-8859-16 should not be allowed when lenient is false
```

**Comment by trantienduchn** [comment]
did you use java 11?

**Comment by chibenwa** [comment]
Yes

**Pull request merged**